torch/testing/_internal/composite_compliance.py

```diff
@@ -173,6 +192,107 @@ def wrap(e):
         tree_map(check, rs)
         return rs
 
+
+def is_tensorlist(lst):
+    if not isinstance(lst, list) and not isinstance(lst, tuple):
+        return False
+    if len(lst) == 0:
+        return False
+    return isinstance(lst[0], torch.Tensor)
```

Comment:
Why not `all(isinstance(t, torch.Tensor) for t in lst)`

Reply:
Sure